lib/_http_agent.js

```diff
@@ -44,10 +44,18 @@ const {
 // ClientRequest.onSocket(). The Agent is now *strictly*
 // concerned with managing a connection pool.
 
+const kReusedHandle = Symbol('kReusedHandle');
 class ReusedHandle {
   constructor(type, handle) {
     this.type = type;
     this.handle = handle;
+    // We need keep the resource object alive from this object, because
+    // domains rely on GC of the resource object for lifetime tracking.
+    // TODO(addaleax): This should really apply to all uses of
+    // AsyncWrap::AsyncReset() when the resource is not the AsyncWrap object
+    // itself. However, HTTPClientAsyncResource and HTTPServerAsyncResource
+    // hold on to other objects, inhibiting GC.
+    handle[kReusedHandle] = this;
```

Comment:
Is it ok that on second reuse of the `handle` the binding to the first `ReusedHandle` is removed?

Reply:
@flarna That only happens together with a `destroy` emitted before and a new ID assigned, so I’m pretty confident it’s safe to discard the first `ReusedHandle` at that point